cruise-control-core/src/main/java/com/linkedin/cruisecontrol/common/Generationed.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+
+package com.linkedin.cruisecontrol.common;
+
+/**
+ * An interface that helps maintain and compare the generation.
+ */
+public interface Generationed<G> {
```

Comment:
Why bother have a parameterized interface and just have objects that have generation numbers associated with them just return their generation number as a long.

Reply:
This is because we have some composite generations. For example, cluster model has a generation which is the combination of metadata generation and the load generation.